core/arch/arm/mm/mobj.c

```diff
@@ -154,9 +156,20 @@ struct mobj *mobj_phys_alloc(paddr_t pa, size_t size, uint32_t cattr,
 		return NULL;
 	}
 
-	va = phys_to_virt(pa, battr);
-	if (!va)
+	/* specific attributes for specific memory objects */
+	switch(battr) {
+	case CORE_MEM_TA_RAM:
+	case CORE_MEM_NSEC_SHM:
+		va = phys_to_virt(pa, battr);
+		if (!va)
+			return NULL;
+		break;
+	case CORE_MEM_SDP_SHM:
+		va = phys_to_virt(pa, battr);
```

Comment:
yes. SDP shm is not default mapped to OP-TEE core, hence does not get a "va".

A `struct mobj` having a NULL `va` field seems valid for a `mobj_phys` object. `mobj_phys_get_va()` will always return `NULL.`